### ISSUE TITLE
Undo issue

### ISSUE THREAD

**Issue by tomitrescak** [xaviergonz/mobx-keystone#115]
Hello, thanks again for addressing my previous issues. I have another small issues with undo. The following code does not properly undo:

```
this.selectedElementRef.current.setSelected(false);
this.selectedElementRef = elementRef(element);
this.selectedElementRef.current.setSelected(true);
```

In the undo operation only the "selected" props of the selected element are undone, but NOT the selectedElementRef. Here is the reproducible case:

https://codesandbox.io/s/empty-browser-kfy1q

To simulate:

1. Click on the "Click to select first"
2. Click on the "Click to select second"
3. Click on undo

Observe how the "selected" state on the individual items was undone but not the selected element.

 It starts to work when you eliminate the first line from the code :/

```
// this.selectedElementRef.current.setSelected(false);
this.selectedElementRef = elementRef(element);
this.selectedElementRef.current.setSelected(true);
```

**Comment by xaviergonz** [comment]
Hi are you sure that's the right code sandbox? I don't see a component to click to undo anywhere there.

**Comment by tomitrescak** [comment]
Sorry, forgot to Save ... should be fine now.

**Comment by xaviergonz** [comment]
Thanks! should be fixed in the just released 0.38.0
